src/java.base/share/classes/sun/security/x509/X509CertImpl.java

```diff
@@ -285,6 +286,7 @@ private DerValue readRFC1421Cert(InputStream in) throws IOException {
      */
     public X509CertImpl(X509CertInfo certInfo) {
         this.info = certInfo;
+        JCAUtil.tryCommitCertEvent(this);
```

Comment:
Why do we need to record this as an event? This is an incomplete (unsigned) certificate.

Reply:
good point. this constructor is mainly used by keytool. Would it make sense if I committed the event after the cert details are signed ?